### PR TITLE
Ensure orchestrators aren't assigned to 32 core machines

### PR DESCRIPTION
**Pull request by zanderso**
https://github.com/flutter/engine/pull/54747 worked, so I'm going to apply that change to all builders on main, and then we can CP it into the release branches.

For https://github.com/flutter/flutter/issues/152186

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -417,12 +476,16 @@ targets:
         {
           "sdk_version": "15a240d"
         }
+    # Do not remove(https://github.com/flutter/flutter/issues/144644)
+    # Scheduler will fail to get the platform
     drone_dimensions:
       - os=Mac-13|Mac-14
 
+  # Avoid using a Mac orchestrator to save ~5 minutes of Mac host time.
   - name: Linux mac_clangd
```
Comment (christopherfujino) [inline .ci.yaml:485]: Should this have cores: "8"?

Comment (zanderso) [reply .ci.yaml:485]: This one doesn't have `release_build: true`, so it won't run on dash-internal. Adding `cores: "8"` would be harmless here, but isn't necessary.

Comment (zanderso) [reply .ci.yaml:485]: Happy to add it everywhere or add more comments? Not sure what would be the most useful.

Comment (christopherfujino) [reply .ci.yaml:485]: ack, this is good as is

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**